### PR TITLE
Catch and log error when refiltering a document

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

**Review by MattiSG — COMMENTED** [review]
This does allow `npm start` to go beyond the refiltering step, but introduces an encoding issue where new versions are created with the wrong encoding when refiltering (and is immediately fixed when updating).
Capture d’écran 2020-09-03 à 11 48 58

File: src/history/recorder.js [hunk 1]
```diff
@@ -76,12 +76,19 @@ export default class Recorder {
       throw new Error(`Only one document of type ${documentType} should have been recorded in ${latestCommit.hash}, but these additional ones have also been recorded: ${filePaths}`);
     }
 
-    const [ recordFilePath ] = filePaths;
+    const [ relativeRecordFilePath ] = filePaths;
+    const recordFilePath = `${this.path}/${relativeRecordFilePath}`;
+    const mimeType = mime.getType(relativeRecordFilePath);
+
+    const readFileOptions = {};
+    if (mimeType != 'application/pdf') {
```
[the remark this answers is not in the document]
Comment (Ndpnt) [reply src/history/recorder.js:84]: 👍

**Review by MattiSG — APPROVED** [review]
Tested locally. This fixes both the refiltering stop and the encoding issue.

File: src/history/recorder.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattiSG) [reply src/history/recorder.js:84]: I'm testing locally with this suggestion.

[the remark this answers is not in the document]
Comment (MattiSG) [reply src/history/recorder.js]: Works fine.

**Pull request merged**